spring-cloud-stream-binder-kafka/src/main/java/org/springframework/cloud/stream/binder/kafka/KafkaMessageChannelBinder.java

```diff
@@ -92,65 +91,38 @@
  */
 public class KafkaMessageChannelBinder extends
 		AbstractMessageChannelBinder<ExtendedConsumerProperties<KafkaConsumerProperties>,
-				ExtendedProducerProperties<KafkaProducerProperties>, Collection<PartitionInfo>, String>
-		implements ExtendedPropertiesBinder<MessageChannel, KafkaConsumerProperties, KafkaProducerProperties>,
-		DisposableBean {
+				ExtendedProducerProperties<KafkaProducerProperties>>
+		implements ExtendedPropertiesBinder<MessageChannel, KafkaConsumerProperties, KafkaProducerProperties> {
 
 	private final KafkaBinderConfigurationProperties configurationProperties;
 
-	private RetryOperations metadataRetryOperations;
-
-	private final Map<String, Collection<PartitionInfo>> topicsInUse = new HashMap<>();
-
 	private ProducerListener<byte[], byte[]> producerListener;
 
-	private volatile Producer<byte[], byte[]> dlqProducer;
-
 	private KafkaExtendedBindingProperties extendedBindingProperties = new KafkaExtendedBindingProperties();
 
-	private AdminUtilsOperation adminUtilsOperation;
+	private RetryOperations metadataRetryOperations;
 
-	public KafkaMessageChannelBinder(KafkaBinderConfigurationProperties configurationProperties) {
-		super(false, headersToMap(configurationProperties));
-		this.configurationProperties = configurationProperties;
-	}
+	private final Map<String, Collection<PartitionInfo>> topicsInUse = new HashMap<>();
 
-	private static String[] headersToMap(KafkaBinderConfigurationProperties configurationProperties) {
-		String[] headersToMap;
-		if (ObjectUtils.isEmpty(configurationProperties.getHeaders())) {
-			headersToMap = BinderHeaders.STANDARD_HEADERS;
-		}
-		else {
-			String[] combinedHeadersToMap = Arrays.copyOfRange(BinderHeaders.STANDARD_HEADERS, 0,
-					BinderHeaders.STANDARD_HEADERS.length + configurationProperties.getHeaders().length);
-			System.arraycopy(configurationProperties.getHeaders(), 0, combinedHeadersToMap,
-					BinderHeaders.STANDARD_HEADERS.length,
-					configurationProperties.getHeaders().length);
-			headersToMap = combinedHeadersToMap;
-		}
-		return headersToMap;
-	}
+	KafkaTopicProvisioner provisioningProvider;
 
-	public void setAdminUtilsOperation(AdminUtilsOperation adminUtilsOperation) {
-		this.adminUtilsOperation = adminUtilsOperation;
+	public KafkaMessageChannelBinder(KafkaBinderConfigurationProperties configurationProperties,
+									KafkaTopicProvisioner provisioningProvider) {
+		super(false, headersToMap(configurationProperties), provisioningProvider);
+		this.configurationProperties = configurationProperties;
+		this.provisioningProvider = provisioningProvider;
 	}
 
 	/**
-	 * Retry configuration for operations such as validating topic creation
 	 *
```

Comment:
I think too that we should move `getPartitionsForTopic()` to the provisioner. Then the binder would have no code to interact with the metadata directly.